pallets/ceres-governance-platform/src/benchmarking.rs

```diff
@@ -30,101 +24,142 @@ fn assert_last_event<T: Config>(generic_event: <T as Config>::RuntimeEvent) {
+        let poll_id = H256::from(encoded);
+        let mut options = BoundedVec::default();
+        options.try_push("Yes".try_into().unwrap()).unwrap();
+        options.try_push("No".try_into().unwrap()).unwrap();
 
         frame_system::Pallet::<T>::inc_providers(&caller);
 
-        let owner: T::AccountId = assets::AssetOwners::<T>::get::<T::AssetId>(CERES_ASSET_ID.into()).unwrap();
+        let owner: T::AccountId = assets::AssetOwners::<T>::get::<T::AssetId>(asset_id.into()).unwrap();
         Assets::<T>::mint(
             RawOrigin::Signed(owner).into(),
             CERES_ASSET_ID.into(),
             caller.clone(),
             number_of_votes
         ).unwrap();
 
-        CeresGovernancePlatform::<T>::create_poll(
-            RawOrigin::Signed(caller.clone()).into(),
-            poll_id.clone(),
-            voting_option,
+        // Create poll
+        let poll_info = PollInfo {
+            poll_asset: asset_id.into(),
             poll_start_timestamp,
-            poll_end_timestamp
-        ).unwrap();
+            poll_end_timestamp,
+            title: title.try_into().unwrap(),
+            description: description.try_into().unwrap(),
+            options,
+        };
+        pallet::PollData::<T>::insert(poll_id, poll_info);
     }: {
         CeresGovernancePlatform::<T>::vote(
             RawOrigin::Signed(caller.clone()).into(),
-            poll_id.clone(),
-            voting_option,
+            poll_id,
+            voting_option.try_into().unwrap(),
             number_of_votes
         ).unwrap();
     }
     verify {
-        assert_last_event::<T>(Event::<T>::Voted(caller, poll_id, voting_option, number_of_votes).into());
+        assert_last_event::<T>(Event::<T>::Voted(caller, poll_id, voting_option.try_into().unwrap(), asset_id.into(), number_of_votes).into());
     }
 
     create_poll {
-        let caller = alice::<T>();
-        let poll_id = Vec::from([1, 2, 3, 4]);
-        let voting_option = 3;
-        let poll_start_timestamp = Timestamp::<T>::get() + 5u32.into();
+        let title = "Title";
+        let description = "Description";
+        let voting_option = "Yes";
+        let asset_id = CERES_ASSET_ID;
+        let number_of_votes = balance!(300);
+        let poll_start_timestamp = Timestamp::<T>::get();
         let poll_end_timestamp = poll_start_timestamp + 10u32.into();
+        let caller = pallet::AuthorityAccount::<T>::get();
+        let nonce = frame_system::Pallet::<T>::account_nonce(&caller);
+        let encoded: [u8; 32] = (&caller, nonce).using_encoded(blake2_256);
+        let poll_id = H256::from(encoded);
+        let mut options = BoundedVec::default();
+        options.try_push("Yes".try_into().unwrap()).unwrap();
+        options.try_push("No".try_into().unwrap()).unwrap();
+
         frame_system::Pallet::<T>::inc_providers(&caller);
+
+        let owner: T::AccountId = assets::AssetOwners::<T>::get::<T::AssetId>(asset_id.into()).unwrap();
```

Comment:
Benchmark should test the worst case. Please set the max possible count for options, max length strings for title and description.
Options also should have names are filled up to the max length.

Reply:
Done.